client.lua

```diff
@@ -89,11 +91,64 @@ local function hasHarness(items)
     harness = _harness
 end
 
+-- Weapon Stress [Haroki]
+local function IsWhitelistedWeaponStress(weapon)
+    if weapon then
+        for _, v in pairs(config.WhitelistedWeaponStress) do
+            if weapon == v then
+                return true
+            end
+        end
+    end
+    return false
+end
+
+local hasWeapon = false
+
+local function startWeaponStressThread(weapon)
+    if IsWhitelistedWeaponStress(weapon) then hasWeapon = false; return end
+
+    CreateThread(function()
+        while hasWeapon do
+            local ped = cache.ped
+            if IsPedShooting(ped) then
+                if math.random() < config.StressChance then
+                    TriggerServerEvent('hud:server:GainStress', math.random(1, 3))
+                end
+            end
+            Wait(0)
+        end
+    end)
+end
+
+AddEventHandler('ox_inventory:currentWeapon', function(currentWeapon)
+    if not currentWeapon then
+        hasWeapon = false
+        return
+    end
```

Comment:
```suggestion
    hasWeapon = false
    Wait(0)

    if not currentWeapon then return end
```
I think it's important to do this to ensure we don't accidentally create another thread without the previous one stopping.